src/UI/Components/Diagram/stencil/stencil.ts

```diff
@@ -0,0 +1,106 @@
+import { dia, ui } from "@inmanta/rappid";
+import { ServiceModel } from "@/Core";
+import { Inventories } from "@/Data/Managers/V2/GETTERS/GetInventoryList";
+import { InstanceStencilTab } from "./instanceStencil";
+import { InventoryStencilTab } from "./inventoryStencil";
+
+/**
+ * Class representing a stencil sidebar.
+ */
+export class StencilSidebar {
+  instanceTab: InstanceStencilTab;
+  inventoryTab: InventoryStencilTab;
+  tabsToolbar: ui.Toolbar;
+
+  /**
+   * Creates a stencil sidebar.
+   *
+   * @param {HTMLElement} stencilElement - The HTML element to which the sidebar elements will be appended.
+   * @param {ui.PaperScroller} scroller - The JointJS scroller associated with the stencil.
+   * @param {Inventories} serviceInventories - The service inventories used to create the inventory stencil tab.
+   * @param {ServiceModel} service - The service model used to create the instance stencil tab.
+   */
+  constructor(
+    stencilElement: HTMLElement,
+    scroller: ui.PaperScroller,
+    serviceInventories: Inventories,
+    service: ServiceModel,
+    serviceModels: ServiceModel[],
+  ) {
+    this.instanceTab = new InstanceStencilTab(
+      stencilElement,
+      scroller,
+      service,
+    );
+    this.inventoryTab = new InventoryStencilTab(
+      stencilElement,
+      scroller,
+      serviceInventories,
+      serviceModels,
+    );
+
+    this.tabsToolbar = new ui.Toolbar({
+      id: "tabs-toolbar",
+      tools: [
+        {
+          type: "button",
+          name: "new_tab",
+          text: "New",
+          id: "new-tab",
+        },
+        {
+          type: "button",
+          name: "inventory_tab",
+          text: "Inventory",
+          id: "inventory-tab",
+        },
+      ],
+    });
+
+    stencilElement.appendChild(this.tabsToolbar.el);
+    this.tabsToolbar.render();
+
+    const firstChild = this.tabsToolbar.el.firstElementChild;
+    const targetElement = firstChild?.firstElementChild;
+
+    //adding active class to the first tab as a default, as Toolbar doesn't apply it when adding 'class' attribute to the tool object
```

Comment:
By that comment I noted that adding `"class": "active"` or `"className": "active" to the Toolbar object didn't work, because to some objects I could add class name on initializiation